### DISCUSSION TITLE
Real-time team chat service?

### DISCUSSION THREAD

**Discussion by SirWumpus** [ioccc-src/mkiocccentry#916] [Ideas]
Other than social media (Mastodon), does the IOCCC support / prefer a common real-time team text chat service?  Wire (E2E, group chats), Discord (pumped IRC), IRC (This Is The Way), other (for lesser mortals)?

Some of the Off-Topic discussions held during an Issue sometimes detract from the actual focus of an issue.  Just wondering if there might be a more suitable channel for history, commentary, general chit-chat, rants & raves, etc.

**Comment by xexyl** [comment]
Nothing of the sort.

I understand the fact there are often OT things brought up. That is in part why there is an OT issue though it's not a perfect solution as sometimes something is relevant to a specific issue.

Email works I guess? IRC is something I have not used in a long time - at least not regularly.

**Comment by SirWumpus** [comment]
![ouija-board-ouija-board-classic-black-and-gold-2](https://github.com/user-attachments/assets/81ef13f8-0970-4f84-8082-e098e0040662)

**Comment by xexyl** [comment]
Funny thing: I clicked the link 'uploading ..' and I guessed that you were making a joke as it just linked to this discussion. Now I see it's an image. I always thought if you clicked comment before it fully uploaded it would fail. Though I never really tried either.

**Comment by lcn2** [comment]
> Some of the Off-Topic discussions held during an Issue sometimes detract from the actual focus of an issue.

Apologies for all of the OT.  No we don't have a a sideband social media platform for that sort of stuff.

We had created an issue for each repo of the form "Question: Issues that aren’t really major issues but are still issues".  The intent was any OT or semi-OT would do there. 

As work winds down we expect those "Question: Issues that aren’t really major issues but are still issues" to be closed. 

Perhaps we should have just opened a discussion for such OT and semi-OT stuff instead of an issue. Would that help, @SirWumpus ?

**Reply by SirWumpus** [reply]
> Perhaps we should have just opened a discussion for such OT and semi-OT stuff instead of an issue. Would that help, @SirWumpus ?

I was thinking that might be the only way to go ultimately, since at the moment its just us three.  But then you might have one such for each repo, which seems like messy overkill.  A real-time side channel probably wouldn't work for those who prefer to work/live/concentrated between rounds of messaging/email vs "always on demand" chat.

Just wondered if there were an alternative, but alas I fear no one solution would be better than what is used now, which is akin to old school dial-up BBS.

**Reply by xexyl** [reply]
> > Perhaps we should have just opened a discussion for such OT and semi-OT stuff instead of an issue. Would that help, @SirWumpus ?
> 
> I was thinking that might be the only way to go ultimately, since at the moment its just us three. But then you might have one such for each repo, which seems like messy overkill. A real-time side channel probably wouldn't work for those who prefer to work/live/concentrated between rounds of messaging/email vs "always on demand" chat.
> 
> Just wondered if there were an alternative, but alas I fear no one solution would be better than what is used now, which is akin to old school dial-up BBS.

Discussions might work although I have always found the issues format more appealing. Even so the problem is there are many cases where an OT comment is warranted in an issue. The OT issue is good but not for everything.

**Reply by SirWumpus** [reply]
> Even so the problem is there are many cases where an OT comment is warranted in an issue.

That should not be a problem, since the RHS Dot Menu on every comment offers the ability to create a new issue referencing the current comment that led up to it being an issue.

**Reply by xexyl** [reply]
> > Even so the problem is there are many cases where an OT comment is warranted in an issue.
> 
> 
> 
> That should not be a problem, since the RHS Dot Menu on every comment offers the ability to create a new issue referencing the current comment that led up to it being an issue.

Creating all those is burdensome though. And it takes it away from the original place.

**Reply by SirWumpus** [reply]
> Creating all those is burdensome though.

An issue per bug or task makes sense. It also allows you to tie commits to specific issues by referencing the issue in the commit message, eg. `GH-nnn blah blah ...`.  Those commits are then cross-referenced in the issue.  If you put minor stuff all under one general issue that never closes, then it even harder to mentally parse.  Its just good developer hygiene.

**Reply by xexyl** [reply]
> > Creating all those is burdensome though.
> 
> 
> 
> An issue per bug or task makes sense. It also allows you to tie commits to specific issues by referencing the issue in the commit message, eg. `GH-nnn blah blah ...`.  Those commits are then cross-referenced in the issue.  If you put minor stuff all under one general issue that never closes, then it even harder to mentally parse.  Its just good developer hygiene.

More things that exist is more to keep track of and we already have a great deal open. 

And it's worked for us fine as is. I don't know but I have so many threads already. Too many to keep track of almost.

**Reply by lcn2** [reply]
> > Creating all those is burdensome though.
> 
> An issue per bug or task makes sense. It also allows you to tie commits to specific issues by referencing the issue in the commit message, eg. `GH-nnn blah blah ...`. Those commits are then cross-referenced in the issue. If you put minor stuff all under one general issue that never closes, then it even harder to mentally parse. Its just good developer hygiene.

Good points.

**Reply by SirWumpus** [reply]
> Too many to keep track of almost.

Probably because they're clustered together under "large" issues / discussions instead of being broken out into smaller more manageable (almost byte size (see what i did there)) tasks.  Its easier to track and close issues as you go and have a better sense of progress (satisfaction) at times.

Granted most of the current work appears to be coming to a conclusion, but something to consider in the future.

**Reply by SirWumpus** [reply]
And there is no reason not to create master tasks that tie a collection of smaller ones together.  Granted the GitHub UI doesn't lend itself well like Phapbricator, but it is possible to [create task lists](https://docs.github.com/en/issues/managing-your-tasks-with-tasklists/creating-a-tasklist) in Markdown.

**Reply by xexyl** [reply]
> > Too many to keep track of almost.
> 
> 
> 
> Probably because they're clustered together under "large" issues / discussions instead of being broken out into smaller more manageable (almost byte size (see what i did there)) tasks.  Its easier to track and close issues as you go and have a better sense of progress (satisfaction) at times.
> 
> 
> 
> Granted most of the current work appears to be coming to a conclusion, but something to consider in the future.  

More is less preferable. The real reason I have so many is not what you say. It's because I have so many pull requests. Each one is another email thread. 

I used to check all but I long since have gone to every one to delete the branch. It's unmanageable and a waste of time. 

I do reply to many but not all. 

Meanwhile the amount of issues that are open isn't very high.

**Reply by xexyl** [reply]
> And there is no reason not to create master tasks that tie a collection of smaller ones together.  Granted the GitHub UI doesn't lend itself well like Phapbricator, but it is possible to [create task lists](https://docs.github.com/en/issues/managing-your-tasks-with-tasklists/creating-a-tasklist) in Markdown.

Sure there is. 

And we already use the task lists you refer to.

**Reply by xexyl** [reply]
> > And there is no reason not to create master tasks that tie a collection of smaller ones together.  Granted the GitHub UI doesn't lend itself well like Phapbricator, but it is possible to [create task lists](https://docs.github.com/en/issues/managing-your-tasks-with-tasklists/creating-a-tasklist) in Markdown.
> 
> Sure there is. 
> 
> And we already use the task lists you refer to. 

There aren't that many OT comments in threads that aren't related and not many OT comments in real issues anyway. 

I don't want to have to go back and forth for the same topic. That's just silly.

**Reply by xexyl** [reply]
> > > And there is no reason not to create master tasks that tie a collection of smaller ones together.  Granted the GitHub UI doesn't lend itself well like Phapbricator, but it is possible to [create task lists](https://docs.github.com/en/issues/managing-your-tasks-with-tasklists/creating-a-tasklist) in Markdown.
> > 
> > Sure there is. 
> > 
> > And we already use the task lists you refer to. 
> 
> There aren't that many OT comments in threads that aren't related and not many OT comments in real issues anyway. 
> 
> I don't want to have to go back and forth for the same topic. That's just silly. 

And also as far as references to issues using the format `#n` is nicer in the sense that it will link to it.  

But that's a stylistic choice I suppose. 

Back another time. Have to go. Good day!

**Reply by SirWumpus** [reply]
> references to issues using the format #n is nicer in the sense that it will link to it.
> But that's a stylistic choice I suppose.

I avoided the use of `#nnn` in commits on favour of `GH-nnn` just in case I end up moving to another service.  The `GH-` prefix at least signals GitHub, maybe GitLab is `GL-` , Sourceforge `SF-`, email-nnn etc.  The `#nnn` is too generic and outside of GitHub, I would not necessarily know where the ticket was reported.  Just a little future proofing my process.

Its not like we're being charged by the byte.

**Reply by xexyl** [reply]
> > references to issues using the format #n is nicer in the sense that it will link to it.
> > But that's a stylistic choice I suppose.
> 
> I avoided the use of `#nnn` in commits on favour of `GH-nnn` just in case I end up moving to another service. The `GH-` prefix at least signals GitHub, maybe GitLab is `GL-` , Sourceforge `SF-`, email-nnn etc. The `#nnn` is too generic and outside of GitHub, I would not necessarily know where the ticket was reported. Just a little future proofing my process.
> 

My immediate and logical[0] reaction would be that you're not elsewhere. But my reasonable reaction that follows right after is that this is a matter of personal preference and style and it does not matter[1]. Sure it helps in GitHub but it's not a requirement. Though I might add that for me GH doesn't necessarily equate to GitHub. I only knew you meant it as you did those PRs on GitHub.

[0] This is not to say that I'm always logical and it's not to say that when I'm logical it's always immediate but it often is. Still I am a human being so there are many times I am not and I also enjoy being silly and even eccentric - in good fun. In this case, as noted, it's not about being logical that matters (something that can sometimes be hard to say).

[1] Do not, however, get me started on the ridiculous warning about misleading indentation in compilers nowadays. That is style policing at best. The same thing for when a `for()` or `while()` loop does not need an actual body of code as the condition/increment does it all, a common C idiom. I have had to put on a semicolon on a line by itself (often) in code just to shut that stupid and useless (and to call it useless is an extreme understatement, the same with the other one about misleading indentation) warning up. When the misleading indentation warning first came to `gcc` it was enabled by `-Wall` (a good one to use) and there was no way to turn the thing off. So I had to 'fix' code that was not broken and not in any way confusing or misleading to anyone who can read C. Changing code on style because a compiler warns about it is nothing but error prone and a waste of time. 

> It's not like we're being charged by the byte.

I suppose we're not. But we might be charged by the bite! On the other hand I also like the `#` form because I follow the `git` best practice of having the first line 51 (I think it was 51 .. maybe a bit more?) chars (or under) and a blank line after it. I configure vim to warn about these things too.

Of course we might ask who gets to decide what is best though (and it should be `better` as there is only one other option but that's another topic entirely). It's in documentation anyway and it is kind of nice to have a shorter summary. Still it can be annoying at times and writing more briefly takes more time, something humorists have pointed out (in amusing ways) and - going back to my favourite, so too did Tolkien (in a letter).

Anyway all that is to say I don't mind if you use your style. If it works for you you should use it!

Still as far as the number of issues I have so many emails from GitHub that it would be even more annoying to have to have another email for the same topic - but I would not want it anyway as I like to keep things together. Sometimes we do take it to the more OT thread though, fwiw.

**Comment by lcn2** [comment]
Any suggestions on such a real-time chat tool, @SirWumpus and if so what are the pluses and minuses of that tool?

**Reply by SirWumpus** [reply]
Well...

* a classic mailing list - everyone has email

* usenet `comp.lang.c.ioccc` maybe (wasn't there an old group long ago?) See https://www.eternal-september.org/ 

* IRC is still used by large projects like OS folks (FreeBSD, NetBSD, and I suppose some Linux distros).  But tends to be "old school". The peach fuzz devs would problem be familiar with Discord.

* [Discord](https://discord.com/) is popular has assorted clients for Mac, Windows, mobile, and web UI (which I use).  A Discord server is easy to create and FREE; can have multiple channels that can come and go as needed; chats, voices, screen sharing (like someone's game or for our purpose maybe group presentation/talks). Typically by invitation, but those are easily given out.  Can be moderated easily.  Allows for image and video uploads (larger file sizes for Nitro "paid" users).  Supports Markdown in chat text; supports security keys like Yubikey.

* Slack is more for business.  These are paid services.  Never seen a free version.  Likewise Teams (barf).

* There are other chat clients, some are mobile centric (I don't own a smart phone), some provide E2EE like Wire which is my favourite (free or paid service available).  But settling on just one would be difficult (iphone vs android, pc vs mac).

Given I've been using Discord for two or three game related "servers" (Buur Pit, Nuclear Option, Super Secret Something, and others) and once setup my own server for a group that I didn't really finish organising, but know how easy it is.  I've seen Discord also used for Play Testers with limited access.

I Discord would be the be most viable place.

**Reply by lcn2** [reply]
Hello @SirWumpus,

> Given I've been using Discord for two or three game related "servers" (Buur Pit, Nuclear Option, Super Secret Something, and others) and once setup my own server for a group that I didn't really finish organising, but know how easy it is.  I've seen Discord also used for Play Testers with limited access.
> 
> 
> 
> I Discord would be the be most viable place.

We created an IOCCC discord server.  Do you want to configure and moderate it?

**Reply by SirWumpus** [reply]
Sure I'll be happy to, my Discord user name is `sir.snert`, add me and give me the necessary permissions and I'll attend to this.  

BTW my Discord display name appears as `[CA] Snert`.  A convention the Buur Pit suggests "[country code] CMDR name".

**Reply by lcn2** [reply]
> Sure I'll be happy to, my Discord user name is `sir.snert`, add me and give me the necessary permissions and I'll attend to this.
> 
> BTW my Discord display name appears as `[CA] Snert`. A convention the Buur Pit suggests "[country code] CMDR name".

Hello @SirWumpus,

Please check your Email for a special 1-time use invite.

## UPDATE 0

Once you have accept the invite, remind us in a comment that you have done so and we will do the next step.

**Reply by SirWumpus** [reply]
PING! Invite complete.

**Reply by xexyl** [reply]
> PING! Invite complete.

I was hoping for IRC .. but as I could not get email very easily or reply or do anything else I guess missed out on my chance. Plus it seems like you're experienced with this and although I have a history with IRC including running my own ircd it's been a long time and I don't have the bandwidth yet for it ... getting FTTH hopefully later this month (if more zoning issues aren't .. issued) but of course in the case of extended power outages or some other thing I couldn't keep it up and running anyway (and I don't want the burden either .. well you get the idea; the point is that we have someone well qualified to help out here which is great! Thanks).

But .. I have never used discord. If it has a macOS client like you say that's good though. The question is ... does this need to go in the FAQ? Also how do I connect etc.?

**Reply by SirWumpus** [reply]
> I was hoping for IRC

Requires hosting, bandwidth, IRC chat clients, some tech. know how.   

> But .. I have never used discord. If it has a macOS client like you say that's good though.

There are several clients available: https://discord.com/download

However I avoid installing extra software and just use the web browser UI, which is equally as capable as a local client.  

> The question is ... does this need to go in the FAQ?

Eventually that will probably be a worth doing, no rush while more pending matters exist.  I think a news announcement (website, github, and Mastodon) will be good starting point; I can do similar on BlueSky, and TwiX (meh).

> Also how do I connect etc

If you forego installing a client in favour of the web UI then [register/login](https://discord.com/login).  I'll send you an invite once the server is ready and open to the public.

@lcn2 I cannot find "IOCCC" when I search Discord.  I assume its not public yet?

**Reply by xexyl** [reply]
> > I was hoping for IRC
> 
> Requires hosting, bandwidth, IRC chat clients, some tech. know how.
> 

True .. like I said even. I just have a lot of history with it is all and I don't have any with Discord. Well we all are fond more of what we're familiar with. I'll worry about this another time though, probably after things are in order for IOCCC28. But we'll see.


> > But .. I have never used discord. If it has a macOS client like you say that's good though.
> 
> There are several clients available: https://discord.com/download
> 

Thanks. That's good to know.

> However I avoid installing extra software and just use the web browser UI, which is equally as capable as a local client.
> 

That's fair enough.

# QUESTION

Do you know if the clients allow logging ? Also, it's just text? Let me rephrase that. There's a text option, right?

> > The question is ... does this need to go in the FAQ?
> 
> Eventually that will probably be a worth doing, no rush while more pending matters exist. I think a news announcement (website, github, and Mastodon) will be good starting point; I can do similar on BlueSky, and TwiX (meh).
> 

That's a probably true yes.

> > Also how do I connect etc
> 
> If you forego installing a client in favour of the web UI then [register/login](https://discord.com/login). I'll send you an invite once the server is ready and open to the public.
> 

Thanks! I'll probably use a client but will worry about it later. As you say there are more urgent matters pending.

> @lcn2 I cannot find "IOCCC" when I search Discord. I assume it's not public yet?

Yes I'd like to know too!

Thanks Anthony, too!

**Reply by SirWumpus** [reply]
> # QUESTION
> 
> Do you know if the clients allow logging ? Also, it's just text? Let me rephrase that. There's a text option, right?

Logging?  As in keeping a transcript?  Messages on Discord are retained.  You can edit and delete your messages.  But otherwise they remain, unlike IRC.

Discord provides text chat channels; voice channels, that can also screen share that might be useful if we have presentations like winners, or talks.  Text is the dominate method, voice channels tend to be used for group events.

> > @lcn2 I cannot find "IOCCC" when I search Discord. I assume it's not public yet?
> 
> Yes I'd like to know too!
> 
> Thanks Anthony, too!

**Reply by xexyl** [reply]
> > # QUESTION
> > Do you know if the clients allow logging ? Also, it's just text? Let me rephrase that. There's a text option, right?
> 
> Logging? As in keeping a transcript? Messages on Discord are retained. You can edit and delete your messages. But otherwise they remain, unlike IRC.
> 

Well I more meant of clients. Like IRC clients can write chats to a log file. I'll find outa anyway if I use it. I was just wondering if you knew. Interesting that it retains the chats though (and indeed unlike with IRC).

> Discord provides text chat channels; voice channels, that can also screen share that might be useful if we have presentations like winners, or talks. Text is the dominate method, voice channels tend to be used for group events.
> 

That's cool.

Thanks!

**Reply by SirWumpus** [reply]
> Well I more meant of clients. Like IRC clients can write chats to a log file. I'll find outa anyway if I use it. I was just wondering if you knew. Interesting that it retains the chats though (and indeed unlike with IRC).

On another Discord server I frequent, I have chats going back at least five years that I can search for.

**Reply by lcn2** [reply]
> @lcn2 I cannot find "IOCCC" when I search Discord. I assume its not public yet?

I should be public .. or is there a setting we forgot to set?

**Reply by SirWumpus** [reply]
So I believe I've done all the setup I can think up and it should all be good (for imaginary values of good).  Things can be twiddled as we progress.  Anyway...

I think we should announce the availablity of the IOCCC DisCord server for disCussions and Chit-Chat.  This is an unlimited no-expiry link to the server.  It can always be revoked later (Server Settings > Invites) if you need to control access. 

        https://discord.gg/NeZV3a57zw

* Put the invite on the web site, post to Mastodon, I'll post to BlueSky, TwiX if you want.
* Add to future IOCCC talk slides with other contact info.
* Put on the IOCCC-SRC account and/or repos.

Personal invites can also be sent with different expiry times: Click the IOCCC tittle > Invite People > "Edit invite link".

If there is no further questions, then I think we can close this DisCussion.

### Update 0
NOTE that server members can hand out Invites.  Right-click on a channel or the server iCon on the LHS in the context menu "Invite people".

**Reply by lcn2** [reply]
Thanks for your work on the Discord server, @SirWumpus 

> ### Update 0
> 
> NOTE that server members can hand out Invites.  Right-click on a channel or the server iCon on the LHS in the context menu "Invite people".
> 
> 

Is allowing others to give invites to others a safe idea?  We are asking about your prior experience in doing that.  What prevents a troll from receiving accepting an invite and then handing out loads in invites to fellow trolls as well to a bunch of "sock accounts" and "bots"?

We always thought of invites as being a privileged operation that is outside the scope of a user.  Maybe that is a wrong assumption on our part.  If so, please examine how users being able to invite anyone else doesn't lead to chaos?

**Reply by xexyl** [reply]
> Thanks for your work on the Discord server, @SirWumpus
> 
> > ### Update 0
> > NOTE that server members can hand out Invites.  Right-click on a channel or the server iCon on the LHS in the context menu "Invite people".
> 
> Is allowing others to give invites to others a safe idea? We are asking about your prior experience in doing that. What prevents a troll from receiving accepting an invite and then handing out loads in invites to fellow trolls as well to a bunch of "sock accounts" and "bots"?
> 
> We always thought of invites as being a privileged operation that is outside the scope of a user. Maybe that is a wrong assumption on our part. If so, please examine how users being able to invite anyone else doesn't lead to chaos?

Kind of like in IRC channel modes where it requires an invite (+i iirc)? I think that's probably a better idea personally for the reasons you cited (which I thought of too). In the IRC only privileged users could give an invite.

But I'm also not experienced with discord so I can't say anything about this specifically: just can say that unfortunately bots and trolls are always a potential problem.

**Reply by lcn2** [reply]
> > Thanks for your work on the Discord server, @SirWumpus
> 
> > 
> 
> > > ### Update 0
> 
> > > NOTE that server members can hand out Invites.  Right-click on a channel or the server iCon on the LHS in the context menu "Invite people".
> 
> > 
> 
> > Is allowing others to give invites to others a safe idea? We are asking about your prior experience in doing that. What prevents a troll from receiving accepting an invite and then handing out loads in invites to fellow trolls as well to a bunch of "sock accounts" and "bots"?
> 
> > 
> 
> > We always thought of invites as being a privileged operation that is outside the scope of a user. Maybe that is a wrong assumption on our part. If so, please examine how users being able to invite anyone else doesn't lead to chaos?
> 
> 
> 
> Kind of like in IRC channel modes where it requires an invite (+i iirc)? I think that's probably a better idea personally for the reasons you cited (which I thought of too). In the IRC only privileged users could give an invite.
> 
> 
> 
> But I'm also not experienced with discord so I can't say anything about this specifically: just can say that unfortunately bots and trolls are always a potential problem.

Oh, we didn't realize that.

We recommend that you remove invite priv from normal users.

## QUESTION 

How does someone request an invite to the server?

On our discord channel "Our Favorite Universe" we hand out invites with a limited lifespan, say on a show we are on.  People who accept the invite and noted by a moderator and can only "see content", they cannot post/create content.  A moderator has to give the newly invited person a special "fan" role for them to be able to post/create content.

That was the concept of our original "ioccc fan" role.  A newly invited person / someone who just accepted an invite link, would have to be elevated from a "read only status" to an "ioccc fan" to post content.

We are not saying this the best way to configure a server mind you .. which is why we are asking.

**Reply by SirWumpus** [reply]
Certainly removing the ability to send invites as @everyone or @fanofioccc can be disabled.  I would allow it for @iocccwinners as I think they make good ambassadors.   I suppose it can be a concern and I think Discord has built-in anti-spam/bot controls.  I've not seen it as an issue with some other servers.

I'd be concerned if lots of people joined that don't appear to participate.   And its possible for Admin to monitor and revoke invites sent.

I still think initially we should have an open invite and see what happens until IOCCC28 closes.  ALSO when when someone registers, maybe include a (time limited) invite with their access tokens.

Example: `Nuclear Option` publishes an open invite in their game that players can click to join.  The player is already a customer and the game is Early Access looking for player feedback, so they might be more liberal.

The `Buur Pit` is more family oriented Elite Dangerous oriented server, so they control diffusion more, through moderators and trusted members and monitor issued invites.

Plus you have at least one moderator who can kick/ban abusive members that get past Discord controls.  

### Quest-ion
Do you have one or two more people who might be regular Discord users from different time zones that can act as Moderators?  Europe and Asia would be good for broad coverage.  Its not critical yet.  Even @xexyl hasn't joined yet and I sent him an invite.

**Reply by xexyl** [reply]
> > > Thanks for your work on the Discord server, @SirWumpus
> > 
> > 
> > > 
> > 
> > 
> > > > ### Update 0
> > 
> > 
> > > > NOTE that server members can hand out Invites.  Right-click on a channel or the server iCon on the LHS in the context menu "Invite people".
> > 
> > 
> > > 
> > 
> > 
> > > Is allowing others to give invites to others a safe idea? We are asking about your prior experience in doing that. What prevents a troll from receiving accepting an invite and then handing out loads in invites to fellow trolls as well to a bunch of "sock accounts" and "bots"?
> > 
> > 
> > > 
> > 
> > 
> > > We always thought of invites as being a privileged operation that is outside the scope of a user. Maybe that is a wrong assumption on our part. If so, please examine how users being able to invite anyone else doesn't lead to chaos?
> > 
> > 
> > Kind of like in IRC channel modes where it requires an invite (+i iirc)? I think that's probably a better idea personally for the reasons you cited (which I thought of too). In the IRC only privileged users could give an invite.
> > But I'm also not experienced with discord so I can't say anything about this specifically: just can say that unfortunately bots and trolls are always a potential problem.
> 
> Oh, we didn't realize that.
> 

You didn't realise IRC had that or something else ?

I used to use IRC a lot. Had an eggdrop bot network with some emechs as well. All for good purposes of course. I wrote some eggdrop scripts as well, some of which emulated some emech features. The eggdrop scripts were in TCL.

So I do have experience with it but I no longer remember raw IRC commands and most of it I have forgotten entirely.


> We recommend that you remove invite priv from normal users.
> 

Although I hardly count in opinion here somehow that seems like a good idea.

> ## QUESTION
> How does someone request an invite to the server?
> 
> On our discord channel "Our Favorite Universe" we hand out invites with a limited lifespan, say on a show we are on. People who accept the invite and noted by a moderator and can only "see content", they cannot post/create content. A moderator has to give the newly invited person a special "fan" role for them to be able to post/create content.
> 
> That was the concept of our original "ioccc fan" role. A newly invited person / someone who just accepted an invite link, would have to be elevated from a "read only status" to an "ioccc fan" to post content.
> 
> We are not saying this the best way to configure a server mind you .. which is why we are asking.

IRC also has this option. Iirc it's +m channel mode: moderated. In that case you have to be given a mode (possibly +v for voice). So it's certainly a valid thought here too.

**Reply by xexyl** [reply]
> I'd be concerned if lots of people joined that don't appear to participate. And its possible for Admin to monitor and revoke invites sent.
> 

On the other hand it depends on what you mean. If you mean lots of people meaning many people who are there and don't participate at all that's good. But I'm not sure how much I would participate: that depends on other things and how busy I am too. There are times too where I don't feel very sociable. But the idea of a chat server for the IOCCC certainly is enticing!


> I still think initially we should have an open invite and see what happens until IOCCC28 closes. ALSO when when someone registers, maybe include a (time limited) invite with their access tokens.
> 

Time limited might be good?


> The `Buur Pit` is more family oriented Elite Dangerous oriented server, so they control diffusion more, through moderators and trusted members and monitor issued invites.

Is that kind of like the instance concept of World of Warcraft (a dear friend had me play for some years years back)?

The MUD I worked on has something like those but years before that other game was even invented. You could also be in NPCs though I had to change some of that as it became a problem when dragons were constantly eating players (I wisely logged it). There was a way to get out (besides another char or chars killing the dragon) but it ended up just being NPCs. But you could also be in objects like a toy box. I meant to do rooms too but I forgot to do that.


> 
> Plus you have at least one moderator who can kick/ban abusive members that get past Discord controls.
> 

Always a good idea.

> ### Quest-ion
> Do you have one or two more people who might be regular Discord users from different time zones that can act as Moderators? Europe and Asia would be good for broad coverage. It's not critical yet. Even @xexyl hasn't joined yet and I sent him an invite.

Oh thanks. How did you send it to me? If it was email I'll have to check. I easily miss emails and I've been quite busy here too (and also having fun with memes). If you did send it by email did you send it to the one in the the JSON file in the IOCCC website ?

I'll certainly join but like I said I'm not sure how often I'd participate. Right now probably not much at all if at all because of all I have to do and get done (and concerned about getting done!). I probably would only participate in between IOCCCs too: during would be something I'd not want to do I would think.

Anyway how did you send the invite? Thanks!

**Reply by SirWumpus** [reply]
@xexyl 
> I probably would only participate in between IOCCCs too: during would be something I'd not want to do I would think.

That's your choice.  I just figure you like to post off-topic stuff a lot on GitHub and now we have a channel for that.

> Anyway how did you send the invite?

I DM it via Mastordon; I didn't have any other private channel (don't know your email).

**Reply by xexyl** [reply]
> @xexyl 
> 
> > I probably would only participate in between IOCCCs too: during would be something I'd not want to do I would think.
> 
> 
> 
> That's your choice.  I just figure you like to post off-topic stuff a lot on GitHub and now we have a channel for that.
> 
> 

Ah. That's possible. Like I said I know nothing about it. 

Of course some of the things I share are primarily and almost exclusively for Landon but posting some programming jokes and other things like it might be fun!

The reason I might not do it during a contest is because I will probably want to work on submissions and assuming I do I wouldn't want to in tiredness accidentally post something revealing: say a clever technique ir something that I thought of. 

That's unlikely: probably wouldn't happen but you never know; it could depend on what others post. 

Well anyway I will see how it goes once things have calmed down more. Thanks for more details!

> 
> > Anyway how did you send the invite?
> 
> 
> 
> I DM it via Mastordon; I didn't have any other private channel (don't know your email).

Ah. That explains it. iOS 18 has an annoying bug (actually a lot of infuriating bugs) where not all notifications are being sent. Of course there's also an annoying lacking feature in the app where unless you are mentioned directly it doesn't show anything. 


I will see about checking in a bit later this week if not today or tomorrow. Thanks!

**Reply by SirWumpus** [reply]
@lcn2 So can I open the IOCCC Discord an publish invites?

**Reply by lcn2** [reply]
> I'd be concerned if lots of people joined that don't appear to participate. And its possible for Admin to monitor and revoke invites sent.
> 
> 

Lots of people might want to just lurk and mostly watch, however.

**Reply by lcn2** [reply]
> Quest-ion
> 
> Do you have one or two more people who might be regular Discord users from different time zones that can act as Moderators? Europe and Asia would be good for broad coverage. Its not critical yet. Even @xexyl hasn't joined yet and I sent him an invite.

Not sure at this point. However, is the surrogates busy I'm sure volunteer moderators could step up.

**Reply by lcn2** [reply]
> @lcn2 So can I open the IOCCC Discord an publish invites?

Soon.  Let's do a review of the setup. 

I'm also considering asking a friend in Canada 🇨🇦 (Nova Scotia) if they would look over the configuration as a super mod and make recommendations as they are experienced with managing discord servers. Not that they would probably stay as an admin, they just have experience in looking at GitHub configuration. 

We should also figure out the workflow of server invites Before it becomes widely open. 

## UPDATE 0

See [GH-discussioncomment-11821826](https://github.com/ioccc-src/mkiocccentry/discussions/916#discussioncomment-11821826) for a question related to "workflow of server invites".

**Reply by SirWumpus** [reply]
> I'm also considering asking a friend in Canada 🇨🇦 (Nova Scotia) if they would look over the configuration as a super mod 

Sounds good. I've only setup 2 small servers (this being one), but a member of several.  I'd welcome feedback.

**Comment by lcn2** [comment]
We invite you to join the IOCCC Discord Community via this link: https://discord.gg/Wa42Qujwnw
